api/nodes/serializers.py

```diff
@@ -935,3 +942,94 @@ def update(self, draft, validated_data):
             draft.update_metadata(metadata)
             draft.save()
         return draft
+
+
+class NodeVOL(ser.Field):
+    def to_representation(self, obj):
+        if obj is not None:
+            return obj._id
+        return None
+
+    def to_internal_value(self, data):
+        return data
+
+
+class NodeViewOnlyLinkSerializer(JSONAPISerializer):
+    key = ser.CharField(read_only=True)
+    id = IDField(source='_id', read_only=True)
+    date_created = ser.DateTimeField(read_only=True)
+    anonymous = ser.BooleanField(required=False, default=False)
+    name = ser.CharField(required=False, default='Shared project link')
+
+    links = LinksField({
+        'self': 'get_absolute_url'
+    })
+
+    creator = RelationshipField(
+        related_view='users:user-detail',
+        related_view_kwargs={'user_id': '<creator._id>'},
+    )
+
+    nodes = RelationshipField(
+        related_view='view-only-links:view-only-link-nodes',
+        related_view_kwargs={'link_id': '<_id>'},
+        self_view='view-only-links:view-only-link-nodes-relationships',
+        self_view_kwargs={'link_id': '<_id>'}
+    )
+
+    def create(self, validated_data):
+        name = validated_data.pop('name')
+        user = get_user_auth(self.context['request']).user
+        anonymous = validated_data.pop('anonymous')
+        node = self.context['view'].get_node()
+
+        try:
+            view_only_link = new_private_link(
+                name=name,
+                user=user,
+                nodes=[node],
+                anonymous=anonymous
+            )
+        except ValidationValueError:
+            raise exceptions.ValidationError('Invalid link name.')
+
+        return view_only_link
+
+    def get_absolute_url(self, obj):
+        node_id = self.context['request'].parser_context['kwargs']['node_id']
+        return absolute_reverse(
+            'nodes:node-view-only-link-detail',
+            kwargs={
+                'link_id': obj._id,
+                'node_id': node_id
+            }
+        )
+
+    class Meta:
+        type_ = 'view_only_links'
+
+
+class NodeViewOnlyLinkUpdateSerializer(NodeViewOnlyLinkSerializer):
+    """
+    Overrides NodeViewOnlyLinkSerializer to not default name and anonymous on update.
+    """
+    anonymous = ser.BooleanField(required=False)
+    name = ser.CharField(required=False)
+
+    def update(self, link, validated_data):
+        name = validated_data.get('name')
+        anonymous = validated_data.get('anonymous')
+
+        assert isinstance(link, PrivateLink), 'link must be a PrivateLink'
+
+        if name:
+            name = strip_html(name)
+            if name is None or not name.strip():
```

Comment:
Should be the only reason name is none is if it's not in validated_data, which means it wouldn't make it to here, right? Or does strip_html sometimes return none? Also, I think we automatically clean incoming data, so the strip_html might not be necessary.